taxcalc/dropq/dropq_utils.py

```diff
@@ -302,59 +299,72 @@ def dropq_summary(df1, df2, mask):
-    itax_diff_bin = create_difference_table(df1, df2,
-                                            groupby='webapp_income_bins',
-                                            income_measure='expanded_income',
-                                            tax_to_diff='iitax')
+    df2['payrolltax'] = df2['payrolltax_xdec']
     ptax_diff_dec = create_difference_table(df1, df2,
                                             groupby='weighted_deciles',
                                             income_measure='expanded_income',
                                             tax_to_diff='payrolltax')
-    ptax_diff_bin = create_difference_table(df1, df2,
-                                            groupby='webapp_income_bins',
-                                            income_measure='expanded_income',
-                                            tax_to_diff='payrolltax')
+    df2['combined'] = df2['combined_xdec']
     comb_diff_dec = create_difference_table(df1, df2,
                                             groupby='weighted_deciles',
                                             income_measure='expanded_income',
                                             tax_to_diff='combined')
+    df2['iitax'] = df2['iitax_xbin']
+    itax_diff_bin = create_difference_table(df1, df2,
+                                            groupby='webapp_income_bins',
+                                            income_measure='expanded_income',
+                                            tax_to_diff='iitax')
+    df2['payrolltax'] = df2['payrolltax_xbin']
+    ptax_diff_bin = create_difference_table(df1, df2,
+                                            groupby='webapp_income_bins',
+                                            income_measure='expanded_income',
+                                            tax_to_diff='iitax')
+    df2['combined'] = df2['combined_xbin']
     comb_diff_bin = create_difference_table(df1, df2,
                                             groupby='webapp_income_bins',
                                             income_measure='expanded_income',
                                             tax_to_diff='combined')
 
-    # create distribution tables
+    # create distribution tables grouped by deciles and bins
     dist1_dec = create_distribution_table(df1, groupby='weighted_deciles',
                                           income_measure='expanded_income',
                                           result_type='weighted_sum')
-    dist2_dec = create_distribution_table(df2, groupby='weighted_deciles',
+    dist1_bin = create_distribution_table(df1, groupby='webapp_income_bins',
                                           income_measure='expanded_income',
                                           result_type='weighted_sum')
-    dist1_bin = create_distribution_table(df1, groupby='webapp_income_bins',
+    for col in [c for c in list(df2) if c.endswith('_xdec')]:
+        df2[col[:-5]] = df2[col]
```

Comment:
This is probably a bit too concise and contains a magic number (why -5?). I think a more maintainable solution would be to:
 - make the list comprehension a local variable with a descriptive name
 - give "5" a descriptive variable name and add a comment on why the last five elements of "col" are chopped off.

Reply:
You're absolutely correct.  Commit 542cedb tries to make things more clear.